pkg/cloudprovider/provider/anexia/network_provisioning.go

```diff
@@ -0,0 +1,122 @@
+package anexia
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/kubermatic/machine-controller/pkg/apis/cluster/common"
+	anxtypes "github.com/kubermatic/machine-controller/pkg/cloudprovider/provider/anexia/types"
+	anxclient "go.anx.io/go-anxcloud/pkg/client"
+	anxaddr "go.anx.io/go-anxcloud/pkg/ipam/address"
+	anxvm "go.anx.io/go-anxcloud/pkg/vsphere/provisioning/vm"
+	"go.uber.org/zap"
+)
+
+func networkInterfacesForProvisioning(ctx context.Context, log *zap.SugaredLogger, client anxclient.Client) ([]anxvm.Network, error) {
+	reconcileContext := getReconcileContext(ctx)
+
+	config := reconcileContext.Config
+	status := reconcileContext.Status
+
+	if status.Networks == nil || len(status.Networks) != len(config.Networks) {
+		if len(status.Networks) != 0 && len(status.Networks) != len(config.Networks) {
+			log.Warn("size of status.Networks != config.Networks, this should not happen in normal operation - ignoring existing status")
+		}
+
+		status.Networks = make([]anxtypes.NetworkStatus, len(config.Networks))
+	}
+
+	ret := make([]anxvm.Network, len(config.Networks))
+	for netIndex, network := range config.Networks {
+		networkStatus := &status.Networks[netIndex]
+		addresses := make([]string, len(network.Prefixes))
+
+		for prefixIndex, prefix := range network.Prefixes {
+			if networkStatus.Addresses == nil {
+				if len(networkStatus.Addresses) != 0 && len(networkStatus.Addresses) != len(network.Prefixes) {
+					log.Warnf("size of status.Networks[%v].Addresses != config.Networks[%v].Prefixes, this should not happen in normal operation - ignoring existing status", netIndex, netIndex)
```

Comment:
A little nitpick, but `%d` fits better here.

```suggestion
					log.Warnf("size of status.Networks[%d].Addresses != config.Networks[%d].Prefixes, this should not happen in normal operation - ignoring existing status", netIndex, netIndex)
```

Reply:
eh, that shouldn't be relevant, `%v` works just as well - using an explicit argument index instead of passing `netIndex` twice on the other hand .. ^^